event.go

```diff
@@ -828,3 +830,21 @@ func (e *Event) MACAddr(key string, ha net.HardwareAddr) *Event {
 	e.buf = enc.AppendMACAddr(enc.AppendKey(e.buf, key), ha)
 	return e
 }
+
+// GetFields returns the JSON decoded fields of the event. Please note that the fields might not be mapped to their original type as JSON decoder maps numbers to float64, etc.
+func (e *Event) GetFields() (map[string]interface{}, error) {
+	if e == nil {
+		return nil, nil
+	}
+
+	eventFields := make(map[string]interface{})
+	buffer := bytes.TrimSpace(e.buf)
+	// If `Msg()` was not called the buffer will be missing the closing curly brace
+	if !bytes.HasSuffix(buffer, []byte("}")) {
```

Comment:
What about JSON defined by an array

```json
[
{"foo":false}
]
```

The last other options for a valid JSON are

```json
true
```

```json
false
```

```json
null
```

But I'm unsure they could happen in your use case

Reply:
Don't think it can happen https://github.com/rs/zerolog/blob/c78e50e2da70f4ae63e1b65222c3acf12e9ba699/log.go#L490 https://github.com/rs/zerolog/blob/c78e50e2da70f4ae63e1b65222c3acf12e9ba699/internal/json/types.go#L18

but that's why I'm returning an error as the parsing can fail for other reasons as well